sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReceiversPumpingScheduler.java

```diff
@@ -0,0 +1,194 @@
+ * <p/>
+ * Note_4: The idle timeout of '60-sec' for the pool and the queue size of '100000' for each Java single-threaded ExecutorService
+ * in the pool are uplifted from the Reactor's choice for the same attributes.
+ */
+public final class ReceiversPumpingScheduler implements Scheduler {
+    private static final String NAME = "receiverPump";
+    private static final int TASK_QUEUE_CAP = 100000;
+    private static final int IDLE_TTL_SECONDS = 60;
+    private static final AtomicReference<ReceiversPumpingScheduler> INSTANCE = new AtomicReference<>();
+    private final Scheduler inner;
+
+    /**
+     * Obtain the common Scheduler.
+     *
+     * @return the Scheduler.
+     */
+    public static Scheduler instance() {
+        if (isVTSchedulerMode()) {
+            return Schedulers.boundedElastic();
+        }
+        ReceiversPumpingScheduler scheduler = INSTANCE.get();
+        if (scheduler != null) {
+            return scheduler;
+        } else {
+            scheduler = new ReceiversPumpingScheduler();
+            if (!INSTANCE.compareAndSet(null, scheduler)) {
+                scheduler.inner.dispose();
+            }
+            return INSTANCE.get();
+        }
+    }
+
+    @Override
+    public Disposable schedule(Runnable task) {
+        return inner.schedule(task);
+    }
+
+    @Override
+    public Disposable schedule(Runnable task, long delay, TimeUnit unit) {
+        return inner.schedule(task, delay, unit);
+    }
+
+    @Override
+    public Disposable schedulePeriodically(Runnable task, long initialDelay, long period, TimeUnit unit) {
+        return inner.schedulePeriodically(task, initialDelay, period, unit);
+    }
+
+    @Override
+    public Worker createWorker() {
+        return inner.createWorker();
+    }
+
+    @Override
+    public long now(TimeUnit unit) {
+        return inner.now(unit);
+    }
+
+    /**
+     *  @deprecated replaced by init.
+     */
+    @Deprecated
+    @Override
+    public void start() {
+        inner.start();
+    }
+
+    @Override
+    public void init() {
+        inner.init();
+    }
+
+    @Override
+    public void dispose() {
+        // NOP as the Scheduler INSTANCE is common.
+    }
+
+    @Override
+    public boolean isDisposed() {
+        return inner.isDisposed();
+    }
+
+    @Override
+    public String toString() {
+        return NAME;
+    }
+
+    private ReceiversPumpingScheduler() {
+        final Supplier<Integer> poolMaxSizeDefault = () -> 20 * Runtime.getRuntime().availableProcessors();
+        // Note: It would be nice to read the custom pool size using com.azure.core.util.Configuration.getGlobalConfiguration.
+        // It requires adding the key to the azure-core known configuration properties, let's evaluate it separately.
+        final Optional<Integer> poolMaxSizeOverridden = Optional.ofNullable(System.getProperty("com.azure.core.amqp.receiversPumpingThreadPoolMaxSize"))
```

Comment:
This Scheduler is like Schedulers._boundedElastic()_ where it’s static in the JVM scope. So, we cannot associate it to a Configuration object scoped to a specific Client builder. We could promote to read from "azure-core" static _Configuration_ object, so it can support sources in addition to `System.getProperty`, I think that can be followed up independently with azure-core team without breaking.

Reply:
This Scheduler is like Schedulers._boundedElastic()_ where it is static in the JVM scope. So, we cannot associate it to a Configuration object scoped to a specific Client builder. We could promote to read from "azure-core" static _Configuration_ object, so it can support all sources not just `System.getProperty`, I think that can be followed up independently with azure-core team without breaking.